File: app/src/main/java/kg/koronastaff/staffapp/adapters/QuarantineAdapter.java
package kg.koronastaff.staffapp.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import kg.koronastaff.staffapp.R;
import kg.koronastaff.staffapp.models.QuarantineSteps;

public class QuarantineAdapter extends RecyclerView.Adapter<QuarantineAdapter.MyViewHolder> {
    private ArrayList<QuarantineSteps> mDataset;

    static class MyViewHolder extends RecyclerView.ViewHolder {
        CardView frame;
        MyViewHolder(CardView v) {
            super(v);
            frame = v;
        }
    }

    public QuarantineAdapter(ArrayList<QuarantineSteps> myDataset) {
        mDataset = myDataset;
    }

    @Override
    public QuarantineAdapter.MyViewHolder onCreateViewHolder(
            ViewGroup parent, int viewType) {

        CardView v = (CardView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.quarantine_steps_item, parent, false);

        MyViewHolder vh = new MyViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        QuarantineSteps f = this.mDataset.get(position);
        TextView title = holder.frame.findViewById(R.id.step_num);
        title.setText(f.getTitle());
        TextView body = holder.frame.findViewById(R.id.step_body);
        body.setText(f.getBody());
    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }

    public void update(ArrayList<QuarantineSteps> myDataset) {
        mDataset = myDataset;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/kg/koronastaff/staffapp/adapters/QuarantineAdapter.kt
package kg.koronastaff.staffapp.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import kg.koronastaff.staffapp.R
import kg.koronastaff.staffapp.models.QuarantineSteps
import java.util.*

class QuarantineAdapter(private var mDataset: ArrayList<QuarantineSteps>) :
        RecyclerView.Adapter<QuarantineAdapter.MyViewHolder>() {

    class MyViewHolder(var frame: CardView) : RecyclerView.ViewHolder(frame)

    override fun onCreateViewHolder(
            parent: ViewGroup, viewType: Int): MyViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.quarantine_steps_item, parent, false) as CardView
        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val f = mDataset[position]
        val title = holder.frame.findViewById<TextView>(R.id.step_num)
        title.text = f.title
        val body = holder.frame.findViewById<TextView>(R.id.step_body)
        body.text = f.body
    }

    override fun getItemCount(): Int {
        return mDataset.size
    }

    fun update(myDataset: ArrayList<QuarantineSteps>) {
        mDataset = myDataset
        notifyDataSetChanged()
    }

}